p2p/src/authenticated/actors/dialer.rs

```diff
@@ -56,11 +59,19 @@ impl<E: Spawner + Clock + GClock + Network + Rng + CryptoRng + Metrics, C: Schem
     }
 
     async fn dial_peers(
-        &self,
+        &mut self,
         tracker: &mut tracker::Mailbox<E, C>,
-        supervisor: &mut spawner::Mailbox<E, SinkOf<E>, StreamOf<E>, C>,
+        supervisor: &mut spawner::Mailbox<E, SinkOf<E>, StreamOf<E>, C::PublicKey>,
     ) {
-        for (peer, address, reservation) in tracker.dialable().await {
+        // Get the list of dialable peers and shuffle to prevent starvation
+        let mut reservations = tracker.dialable().await;
+        reservations.shuffle(&mut self.context);
+
+        for res in reservations {
+            let ResMetadata::Dialer(peer, address) = res.metadata().clone() else {
+                panic!("unexpected reservation type");
```

Comment:
Suggest using `unreachable!` and adding a comment as to why this should never not be `Dialer`?